app/src/main/kotlin/com/mapbox/vision/teaser/BaseVisionFragment.kt

```diff
@@ -0,0 +1,39 @@
+package com.mapbox.vision.teaser
+
+import android.os.Bundle
+import android.view.View
+import androidx.fragment.app.Fragment
+import com.mapbox.vision.mobile.core.models.Country
+import com.mapbox.vision.teaser.utils.classification.SignResources
+import com.mapbox.vision.teaser.utils.dpToPx
+
+open class BaseVisionFragment : Fragment() {
+
+    protected var country = Country.Unknown
+    protected var calibrationProgress = 0F
+    protected var lastSpeed = 0f
+
+    protected var signSize = 0
```

Comment:
Do we need it in Base fragment? Seems like a specific SignDetection property

Reply:
Same applies for country

Reply:
The country is used in Safety and Sign
signSize and margin are used in Lane and Sign

I propose for now to use common BaseFragment to update some params (even if it is actually for one fragment but redundant to other fragments). BaseFragment allows also share some common code and avoids multiple interfaces for a single action. Generally, less code and abstractions around simple logic.

But probably you have better ideas

Reply:
Ah, right, don't mind sharing properties that are used in several places.
While it's true for country/margin though, is `signSize` though really used in Lane? 
Also `margin` declared in a separate BaseFragment seems not very clear, e.g. what margin is it for? IMHO visual properties can be declared private separately in each fragment that requires them

Reply:
Okay, I'll put signSize and margin properties to own fragments

Reply:
fixed